### DISCUSSION TITLE
Add new awaited keyword to allows better chaining.

### DISCUSSION THREAD

**Discussion by cafenroy** [dotnet/csharplang#6459] [Language Ideas] [answered]
I know the Rust like **.await** suggestion has been rejected. But the idea of better chaining for **await** is still considered.

I would go with a new keyword name **awaited** that could be written at the end instead of the begining.

`var result = (await query.ToListAsync()).GroupBy(...)`

could also be written like this: 

`var result = query.ToListAsync() awaited.GroupBy(...)`

**Comment by HaloFour** [comment, marked as answer]
Not quite the same, but I had suggested this: #4076

The team had [this feedback](https://github.com/dotnet/csharplang/blob/main/meetings/2020/LDM-2020-11-11.md#add-await-as-a-dotted-postfix-operator) about it:

> A more general approach that simplified chaining generally for prefix operators would be more interesting.

**Reply by cafenroy** [reply]
I saw your suggestion. Mine is to add a new keyword that wouldn't need a dot. It is very similar indeed.

**Comment by qrli** [comment]
A workaround using existing feature:
```cs
var result = await query.ToListAsync() switch { var x => x.GroupBy(...) };
```

**Reply by qrli** [reply]
Ideally, I'd like something like:
```cs
var result = await query.ToListAsync() then GroupBy(...).Where(...).Whatever(...);
```
where `then` is an operator with lowest precedence.

**Reply by cafenroy** [reply]
The suggestion in this discussion is to add an `awaited` keyword that can be appended at the end of the expression.

**Reply by qrli** [reply]
Yes, I know. But as mentioned in HaloFour's post, the LDM preferred something which can be applicable to all prefix operators, instead of a specific solution for `await` only.
